File: ch6/6.3.2/ObjC/CustomCell/CustomCell/CustomCell.m
//
//  CustomCell.m
//  CustomCell
//
//  Created by tonyguan on 2016/10/24.
//  Copyright © 2016年 关东升. All rights reserved.
//

#import "CustomCell.h"

@implementation CustomCell

-(instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    
    if (self) {
        
        //单元格的宽度
        //CGFloat cellWidth = self.frame.size.width;
        CGFloat cellHeight = self.frame.size.height;
        
        CGFloat imageViewWidth = 39;
        CGFloat imageViewHeight = 28;
        CGFloat imageViewLeftView = 300;
        
        /// 1.添加ImageView
        self.myImageView = [[UIImageView alloc] initWithFrame: CGRectMake(imageViewLeftView,  (cellHeight - imageViewHeight) / 2, imageViewWidth, imageViewHeight)];
        
        [self addSubview:self.myImageView];
        
        /// 2.添加标签
        CGFloat labelWidth = 120;
        CGFloat labelHeight = 21;
        CGFloat labelLeftView = 15;
        self.myLabel = [[UILabel alloc] initWithFrame:CGRectMake(labelLeftView, (cellHeight - labelHeight) / 2, labelWidth, labelHeight)];
        
        [self addSubview:self.myLabel];
    }
    
    return self;
}

@end

File: ch6/6.3.2/Swift/CustomCell/CustomCell/CustomCell.swift
//
//  CustomCell.swift
//  CustomCell
//
//  Created by tonyguan on 2016/10/24.
//  Copyright © 2016年 关东升. All rights reserved.
//

import UIKit

class CustomCell: UITableViewCell {

    var myLabel: UILabel!
    var myImageView: UIImageView!
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        //单元格的宽度
        //let cellWidth: CGFloat = self.frame.size.width
        let cellHeight: CGFloat = self.frame.size.height
        
        let imageViewWidth: CGFloat = 39
        let imageViewHeight: CGFloat = 28
        let imageViewLeftView: CGFloat = 300
        
        
        /// 1.添加ImageView
        self.myImageView = UIImageView(frame: CGRect(x: imageViewLeftView,  y: (cellHeight - imageViewHeight) / 2, width: imageViewWidth, height: imageViewHeight))
        
        self.addSubview(self.myImageView)
        
        /// 2.添加标签
        let labelWidth: CGFloat = 120
        let labelHeight: CGFloat = 21
        let labelLeftView: CGFloat = 15
        
        self.myLabel = UILabel(frame: CGRect(x: labelLeftView, y: (cellHeight - labelHeight) / 2, width: labelWidth, height: labelHeight))
        
        self.addSubview(self.myLabel)
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
